src/Grpc.Net.Client/Internal/Retry/RetryCallBase.cs

```diff
@@ -82,7 +82,7 @@ protected RetryCallBase(GrpcChannel channel, Method<TRequest, TResponse> method,
                 _ctsRegistration = RegisterRetryCancellationToken(options.CancellationToken);
             }
 
-            var deadline = Options.Deadline.GetValueOrDefault(DateTime.MaxValue);
+            var deadline = Options.Deadline ?? DateTime.MaxValue;
```

Comment:
```suggestion
            var deadline = Options.Deadline.GetValueOrDefault(DateTime.MaxValue);
```
The new code [produces](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8aHunGBLAHYYGAWQAUgjAH4GCAJQMAvAD5ZDKTJSkA3Fx70GkhgDkJQmfKWqEAOgDiMDADVsAGwCuMAPJQAIjAAzbHdXDDEtOV1qAF8gA=) more IL and machine code. I'd keep the previous state.